packages/core/contracts/common/implementation/FixedPoint.sol

```diff
@@ -373,4 +377,386 @@ library FixedPoint {
             output = mul(output, a);
         }
     }
+
+    // ------------------------------------------------- SIGNED -------------------------------------------------------------
+    // Supports 18 decimals. E.g., 1e18 represents "1", 5e17 represents "0.5".
+    // For signed values:
+    //   This can represent a value up (or down) to +-(2^255 - 1)/10^18 = ~10^58. 10^58 will be stored internally as int256 10^76.
+    int256 private constant SFP_SCALING_FACTOR = 10**18;
+
+    struct Signed {
+        int256 rawValue;
+    }
+
+    function fromSigned(Signed memory a) internal pure returns (Unsigned memory) {
+        require(a.rawValue >= 0, "Negative value provided");
+    }
+
+    function fromUnsigned(Unsigned memory a) internal pure returns (Signed memory) {
+        require(a.rawValue <= uint256(type(int256).max), "Unsigned too large");
+        return Signed(int256(a.rawValue));
+    }
```

Comment:
Note: these functions are here to allow converting back and forth between signed and unsigned values while checking bounds. We might consider changing their names if they seem confusing.